### PR TITLE
Added unit tests for config_parser

### PR DESCRIPTION
**Pull request by jasonsliu**

### CODE REVIEW

File: config_parser_test.cc [hunk 1]
```diff
@@ -78,3 +78,31 @@ TEST_F(NginxStringConfigTest, QuoteConfig) {
...
+
+TEST(ServerInfoTest, UnexpectedConfig) {
+	ServerInfo info;
+	EXPECT_FALSE(getServerInfo("test_config_bad", &info));
+}
+
+TEST(ServerInfoTest, UnexpectedHandlerConfig) {
+	ServerInfo info;
+	EXPECT_FALSE(getServerInfo("test_config_bad2", &info));
+}
```
Comment (crystalhsieh) [inline config_parser_test.cc]: Please add an extra line at the bottom :)

File: config_parser_test.cc [hunk 2]
```diff
@@ -78,3 +78,31 @@ TEST_F(NginxStringConfigTest, QuoteConfig) {
 TEST_F(NginxStringConfigTest, InvalidQuoteConfig) {
 	EXPECT_FALSE(ParseString("foo \"bar;"));
 }
+
+TEST_F(NginxStringConfigTest, NginxConfigToString) {
+	EXPECT_TRUE(ParseString("boo far;"));
+	EXPECT_EQ(out_config_.ToString(), "boo far;\n");
+}
+
+TEST_F(NginxStringConfigTest, StatmentToString) {
```
Comment (crystalhsieh) [inline config_parser_test.cc]: Change to "StatementToString" please!

File: config_parser_test.cc [hunk 3]
```diff
@@ -78,3 +78,31 @@ TEST_F(NginxStringConfigTest, QuoteConfig) {
...
+	EXPECT_TRUE(ParseString("boo far;"));
+	EXPECT_EQ(out_config_.statements_[0]->ToString(0), "boo far;\n");
+}
+
+TEST(ServerInfoTest, GetServerInfo) {
+	ServerInfo info;
+	EXPECT_TRUE(getServerInfo("test_config", &info));
+	EXPECT_EQ(info.port, 1890);
+	EXPECT_EQ(info.echo_path_to_root_.size(), 1);
```
Comment (crystalhsieh) [inline config_parser_test.cc:96]: can you also add a test that checks that it maps to the correct value?

**Review by ryanvoong — APPROVED** [review]
looks good

**Pull request merged**